sympy/controlsys/lti.py

```diff
@@ -0,0 +1,757 @@
+
+        The parallel interconnection of two systems P1 and P2 is the system for which
+        u = u1 + u2 and y = y1 + y2 so that:
+
+                  ----  y1
+             --> | P1 |---
+            |     ----    |+
+        u --|             o ---> y
+            |     ----    |+
+             --> | P2 |---
+                  ----  y2
+
+        Parameters
+        ==========
+
+        anotherSystem : StateSpaceModel
+            StateSpace representation of the model you want to interconnect with
+            the current model
+
+        See Also
+        ========
+
+        cascade: cascade interconnection of two systems
+        """
+        if not isinstance(anotherSystem, StateSpaceModel):
+            raise TypeError("Argument must be of type StateSpaceModel, not %r" % (type(anotherSystem)))
+        # assert matching shapes
+        if not ((self.represent[1].shape[1] == anotherSystem.represent[1].shape[1]) and
+                (self.represent[2].shape[0] == anotherSystem.represent[2].shape[0])):
+            raise ShapeError("Dimensions of inputs and outputs must match!")
+
+        newA = self.represent[0].col_join(zeros(self.represent[0].rows, anotherSystem.represent[0].cols)) \
+                                .row_join(
+                                    zeros(anotherSystem.represent[0].rows, self.represent[0].cols)
+                                    .col_join(anotherSystem.represent[0]))
+        newB = self.represent[1].col_join(anotherSystem.represent[1])
+        newC = self.represent[2].row_join(anotherSystem.represent[2])
+        newD = self.represent[3] + anotherSystem.represent[3]
+
+        return StateSpaceModel([newA, newB, newC, newD])
+
+    #
+    # define a magic function for unknown method handling
+    #   the class tries to pass the method to the matrices in self.represent
+    def __getattr__(self, name):
+
+        # dont overwrite private or magic function attribute testing!
+        if name[0] == '_':
+            raise AttributeError("%r object has no attribute %r" %
+                                 (self.__class__, name))
+
+        try:
+            def handler(*args, **kwargs):
+
+                new_represent = []
+                for r in self.represent:
+                    methodToCall = getattr(r, name)
+                    new_represent.append(methodToCall(*args, **kwargs))
+                return StateSpaceModel(new_represent)
+
+        except AttributeError:
+            raise AttributeError("%r object has no attribute %r" %
+                                 (self.__class__, name))
+
+        return handler
+
+    #
+    # _repr_latex_(self)
+    #   defines the representation of the class in ipython pretty printing
+    #
+    def _repr_latex_(self):
+        return '$' + latex(self.BlockRepresent) + '$'
+
+    def __str__(self):
+        return 'StateSpaceModel' + str(self.BlockRepresent)[6:]
+
+
+class TransferFunctionModel(object):
```

Comment:
Since a transfer function is a ratio of two polynomials in s you will likely benefit from subclassing from a core sympy type here.